### PR TITLE
Implement exporting Dread games via FTP

### PR DESCRIPTION
**Pull request by henriquegemignani**

### CODE REVIEW

File: randovania/games/dread/gui/dialog/ftp_uploader.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        with FTP() as ftp:
+            ftp.connect(self.ip, self.port)
+            if self.auth is not None:
+                ftp.login(*self.auth)
+            else:
+                ftp.login()
+
+            # Delete any existing remote path
+            delete_path(ftp, self.remote_path, lambda name: progress_update(f"Deleting existing file: {name}", 0))
```
Comment (duncathan) [inline randovania/games/dread/gui/dialog/ftp_uploader.py:53]: I'd rather this use `progress_update(str, -1)` instead, if it's not gonna be updating the progress value

File: randovania/games/dread/gui/dialog/game_export_dialog.py [hunk 2]
```diff
@@ -182,7 +212,7 @@ def _on_update_target_platform(self):
         )
         self.output_tab_widget.setTabVisible(
             self.output_tab_widget.indexOf(self.tab_ftp),
-            target_platform == DreadModPlatform.ATMOSPHERE and False
+            target_platform == DreadModPlatform.ATMOSPHERE
```
Comment (duncathan) [inline randovania/games/dread/gui/dialog/game_export_dialog.py:215]: maybe check for mod manager too?

File: randovania/gui/ui_files/dread_game_export_dialog.ui [hunk 3]
```diff
@@ -24,7 +24,7 @@
    <item row="7" column="0" rowspan="2" colspan="3">
     <widget class="QTabWidget" name="output_tab_widget">
      <property name="currentIndex">
-      <number>0</number>
+      <number>1</number>
```
Comment (duncathan) [inline randovania/gui/ui_files/dread_game_export_dialog.ui:27]: intentional?

**Review by duncathan — CHANGES_REQUESTED** [review]

File: randovania/games/dread/gui/dialog/game_export_dialog.py (hunk 2 above)
Comment (henriquegemignani) [reply randovania/games/dread/gui/dialog/game_export_dialog.py:215]: mod manager is a tab-specific setting that only exists on the "SD" tab.

File: randovania/gui/ui_files/dread_game_export_dialog.ui (hunk 3 above)
Comment (henriquegemignani) [reply randovania/gui/ui_files/dread_game_export_dialog.ui:27]: or, I forgot about this

File: randovania/games/dread/gui/dialog/game_export_dialog.py (hunk 2 above)
Comment (duncathan) [reply randovania/games/dread/gui/dialog/game_export_dialog.py:215]: in that case, let's just make sure that it's clear to the user that FTP requires mod manager. I didn't look at the actual UI so I'm not sure if it does that already

Comment (henriquegemignani) [reply randovania/games/dread/gui/dialog/game_export_dialog.py:215]: image

**Pull request merged**